Domain.Sql/ExclusiveEventStoreCatchupQuery.cs

```diff
@@ -67,6 +68,7 @@ internal class ExclusiveEventStoreCatchupQuery : IDisposable
                 eventQuery = eventQuery.Take(batchSize);
 
                 events = DurableStreamFrom(eventQuery, startAtId);
+                dbContext.Database.CommandTimeout = oldCommandTimeout;
```

Comment:
Is the intention to prevent the `Count` operation from timing out or `Take` operation as well?

Reply:
If the intent is to prevent `Count` from timing out, it would be safer to move the resetting of the `CommandTimeout` into a `finally` block after `Count` is performed.